tests/integration/ember-engines-test.js

```diff
@@ -0,0 +1,50 @@
+import { module, test } from 'qunit';
+import {
+  setupContext,
+  setupRenderingContext,
+  teardownContext,
+  teardownRenderingContext,
+  render,
+} from '@ember/test-helpers';
+import engineResolverFor from 'ember-engines/test-support/engine-resolver-for';
+import hbs from 'htmlbars-inline-precompile';
+
+const modulePrefix = 'eager-engine';
+const resolver = engineResolverFor(modulePrefix);
+
+module('setupRenderingContext for "ember-engines"', function (hooks) {
+  hooks.beforeEach(async function () {
+    await setupContext(this, { resolver });
+    // this.owner now is an engine.
+    this.engine = this.owner;
+    await setupRenderingContext(this);
+  });
+
+  hooks.afterEach(async function () {
+    await teardownRenderingContext(this);
+    this.engine = undefined;
+    await teardownContext(this);
+  });
+
+  test('render components from engines', async function (assert) {
+    assert.expect(2);
+
+    this.set('colorValue', 'red');
+
+    await render(hbs`{{pretty-color name=colorValue}}`);
+
+    assert.equal(
+      this.element.querySelector('div').getAttribute('style'),
+      'color: red',
+      'starts as red'
+    );
+
+    this.set('colorValue', 'blue');
+
+    assert.equal(
+      this.element.querySelector('div').getAttribute('style'),
+      'color: blue',
+      'updates to blue'
+    );
+  });
```

Comment:
This test doesn't _really_ work (it is just using `this.owner`), we should have a test that registers a component into `this.engine` (when `this.engine !== this.owner`) that is not present on `this.owner` and render it.

Reply:
In this case `this.owner` is an engine using `engineResolverFor`(that's what this helper does). And this component just exist in the `eager-engine`. then it's working.

Reply:
makes sense!

do you want something like this

https://github.com/emberjs/ember.js/blob/f73d8440d19cf86a10c61ddb89d45881acfcf974/packages/%40ember/-internals/routing/tests/ext/controller_test.js#L15-L23